cmd/antrea-agent/agent.go

```diff
@@ -206,6 +207,14 @@ func run(o *Options) error {
 	// exits, we will force exit.
 	stopCh := signals.RegisterSignalHandlers()
 
+	// Start the NPL agent.
+	if features.DefaultFeatureGate.Enabled(features.NodePortLocal) {
+		err := npl.InitializeNPLAgent(k8sClient, informerFactory, o.config.NPLPortRange)
+		if err != nil {
+			klog.Warningf("Failed to start NPL agent: %v", err)
```

Comment:
Should the process exit with error if it fails to initialize a requested feature?

Reply:
Makes Sense. I'll add it.